guardrails/validators.py

```diff
@@ -448,63 +450,6 @@ def to_prompt(self, with_keywords: bool = True) -> str:
         return self.field_validator.__func__.__name__
 
 
-@register_validator(name="choice", data_type="choice")
-class Choice(Validator):
```

Comment:
Why was this validator removed?

Reply:
Skeleton validation covers what the Choice validator was doing

Reply:
Rather than dropping it, which would result in a `UserWarning: Validator choice is not valid for element...`, it would be a better experience to disable its functionality and add a deprecation warning explaining that this now happens elsewhere plus any standard language around `this will be removed in the next major version` etc..

Reply:
The Choice validator was only meant to be used by the Choice datatype, but we realized that it just duplicates skeleton validation. The motivation for its removal is because it's incompatible with the new validator interface; it relies on accessing the object outside of the field being validated.